src/generic_ui/polymer/tip.ts

```diff
@@ -0,0 +1,64 @@
+Polymer({
+  close: function() {
+    this.fire('closed');
+    this.remove();
+  },
+  doReposition: function() {
+    var prev = this.previousSibling;
+    if (prev.offsetLeft === 0 && prev.offsetTop === 0 && prev.offsetWidth === 0
+        && prev.offsetHeight === 0) {
+      /* if the DOM is not set up yet, keep checking every half second */
```

Comment:
I wasn't able to find anything for figuring out when the element is actually displayed to the user (the way we keep elements hidden means that they will not have the position set even when any of the normal events would be trtiggered).